snippets/csharp/programming-guide/strings/string_program_1.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+
+namespace StringCompare
+{
+    class Program
+    {
+        static void Main()
+        {
+            string jupiterFirst = "Jupiter";
+            string jupiterLast = "Hammon";
+            int jupiterBorn = 1711;
+            int jupiterPub = 1761;
+            // Variables
+            Console.WriteLine($"{jupiterFirst} {jupiterLast} was an African American poet born in {jupiterBorn}.");
+            // Simple expression
+            Console.WriteLine($"He was first published in {jupiterPub} at the age of {jupiterPub - jupiterBorn}.");
+            // Complex Expressions
+            Console.WriteLine($"He'd be over {Math.Round((2018d - jupiterBorn) / 100d, 0) * 100d} years old today.");
+
+            // Output:
+            // Jupiter Hammon was an African American poet born in 1711.
+            // He was first published in 1761 at the age of 50.
+            // He'd be over 300 years old today. 
+
+            System.Console.WriteLine();
+
+            string phillisFirst = "Phillis";
+            string phillisLast = "Wheatley";
+            int phillisBorn = 1753;
+            int phillisPub = 1773;
```

Comment:
@BillWagner would it be a good case to use named tuples here?
https://docs.microsoft.com/en-us/dotnet/csharp/tuples#named-and-unnamed-tuples

```csharp
var phillisWheatley = (firstName: "Phillis", lastName: "Wheatley", born: 1753, firstPublished: 1773);
```

Reply:
Thanks @pkulikov. Yup! That's the [#Code link](https://docs.microsoft.com/en-us/contribute/dotnet-style-guide#code) I needed. Ya know..the one with with all the answers! I guess I have a lot more to learn about markdown and snippets, which is great! I'm going to start working through these changes tomorrow. Really enjoying this. I'll wait to see any additional feedback from @BillWagner prior. Thanks.

Reply:
This ValueTuple suggestion is fantastic and I implemented it gladly. Knocked all the declarations to a single line. I had to get a NuGet package to support the feature on windows. It ran right away on my mac. But what an awesome new feature! Thanks for this.

Reply:
> I had to get a NuGet package to support the feature on windows. It ran right away on my mac.

Sounds strange to me. If you run it on .NET Core 2.1 (as the project file says), should be the same experience on both Windows and Mac.

Reply:
When I saw this line I just dropped it into an available .Net Framework console app on a windows machine to play with and learn about, then installed the package. The project file is housed on mac as a Core app, there the feature integrated seamlessly. Would this explain it?

Reply:
Also, would it be helpful to include a comment for users unfamiliar with this new feature? 
// Declare Named Tuple
Of course, it won't translate and can't link. Alternatively, I could include a link in the lead up to snippet. Ie. In this example of string formatting...we declare variables using a [named tuple}(https://docs.microsoft.com/en-us/dotnet/csharp/tuples). Thoughts? Or just leave it out...